### PR TITLE
Add Bridge Fee to Move Bridge Module

### PR DESCRIPTION
**Pull request by Primata**
## Description

Single sided bridge fee on the move side. Relayer has to maintain the fee price.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [x] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
Untested

## Key Areas to Review
Initial fee and if this is viable.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by andygolay** [comment]
@Primata Not sure if maybe you want to have this go into the `movement` branch, also:

- How is the fee tested?
- Is there an issue this can be pinned to?

**Comment by Primata** [comment]
> @Primata Not sure if maybe you want to have this go into the `movement` branch, also:
> 
> 
> 
> - How is the fee tested?
> 
> - Is there an issue this can be pinned to? 

@andygolay this was one of the three proposals on how to handle bridge fees. It's standing here as just a possible path but it seems like we won't take it.

File: aptos-move/framework/aptos-framework/sources/native_bridge.move [hunk 1]
```diff
@@ -185,16 +197,92 @@ module aptos_framework::native_bridge {
...
+    /// Simplified version of complete_bridge_transfer
+    ///  
+    /// @param caller The signer representing the bridge relayer.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.    
+    /// @abort If the caller is not the bridge relayer or the transfer has already been processed.  
+    public entry fun complete_bridge_transfer_v2(
```
Comment (andygolay) [inline aptos-move/framework/aptos-framework/sources/native_bridge.move:210]: What is this `complete_bridge_transfer_v2`?

Looks like there's no parameter check. 

Also v2 isn't descriptive... just confused about what the purpose is of this function. We shouldn't just throw in functions without a good motivation.

**Review by andygolay — CHANGES_REQUESTED** [review]

**Review by andygolay — CHANGES_REQUESTED** [review]
Other than my comments about that `_v2` function, I'm wondering, is there an issue or an MIP associated with this PR? We're trying to build in a more organized way, so we can track PRs with issues. I looked in `aptos-core` and `movement` and didn't see an issue suggesting we should add such a fee. 

I think I recall some discussion on Slack, but that's not the same as a documented issue or formal discussion with a paper trail we can use to justify developments to auditors (and for sanity, so we are conscious of what we're doing and can point to formal decision-making processes as the reasons for making the changes we've made). 

If there's any relevant issues, MIPs, or discussions to this PR, could you please link it / mention it in the PR description?

**Comment by andygolay** [comment]
- `movement move test` fails with

```
error[E10005]: unable to generate test
    ┌─ /Users/andygmove/Downloads/repos/aptos-core/aptos-move/framework/aptos-framework/sources/native_bridge.move:218:7
    │
218 │     #[test(aptos_framework = @aptos_framework, sender = @0x726563697069656e740000000000000000000000000000000000000000000000)]
    │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Missing test parameter assignment in test. Expected a parameter to be assigned in this attribute
219 │     fun test_initiate_bridge_transfer_happy_path(
    │         ---------------------------------------- Error found in this test
    ·
222 │         relayer: &signer
    │         ------- Corresponding to this parameter
```

This also causes the E2E Move test `test_native_bridge_initiate` to fail:

```
failures:

---- tests::bridge::test_native_bridge_initiate stdout ----
thread 'tests::bridge::test_native_bridge_initiate' panicked at aptos-move/e2e-move-tests/src/harness.rs:422:9:
assertion `left == right` failed
  left: Keep(MoveAbort { location: 0000000000000000000000000000000000000000000000000000000000000001::native_bridge, code: 5, info: Some(AbortInfo { reason_name: "EINVALID_VALUE", description: "" }) })
 right: Keep(Success)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    tests::bridge::test_native_bridge_initiate
```

**Review by andygolay — DISMISSED** [review]
LGTM

**Review by andygolay — APPROVED** [review]

**Comment by 0xmovses** [comment]
We're unable to merge this because of unsigned commits.

**Comment by Primata** [comment]
coming from @franck44

**Review by radupopa369 — APPROVED** [review]

**Pull request merged**